.travis.yml

```diff
@@ -46,174 +43,256 @@ cache:
+                              gcc, ghc-7.8.4], sources: [hvr-ghc, ubuntu-toolchain-r-test]}}
 
   - env: BUILD=stack ARGS="--resolver lts-3"
+    compiler: ": #stack 7.10.2"
     addons: {apt: {packages: [libgmp-dev,
                               libmpfr-dev,
                               libmpc-dev,
                               binutils-dev,
                               g++-4.7,
-                              gcc,
-                              ghc-7.10.2],
-                  sources: [hvr-ghc, ubuntu-toolchain-r-test]}}
+                              gcc, ghc-7.10.2], sources: [hvr-ghc, ubuntu-toolchain-r-test]}}
 
   - env: BUILD=stack ARGS="--resolver lts-5"
+    compiler: ": #stack 7.10.3"
     addons: {apt: {packages: [libgmp-dev,
                               libmpfr-dev,
                               libmpc-dev,
                               binutils-dev,
                               g++-4.7,
-                              gcc,
-                              ghc-7.10.3],
-                  sources: [hvr-ghc, ubuntu-toolchain-r-test]}}
+                              gcc, ghc-7.10.3], sources: [hvr-ghc, ubuntu-toolchain-r-test]}}
 
   # Nightly builds are allowed to fail
   - env: BUILD=stack ARGS="--resolver nightly"
+    compiler: ": #stack nightly"
     addons: {apt: {packages: [libgmp-dev,
                               libmpfr-dev,
                               libmpc-dev,
                               binutils-dev,
                               g++-4.7,
-                              gcc,
-                              ghc-7.10.3],
-                  sources: [hvr-ghc, ubuntu-toolchain-r-test]}}
+                              gcc, libgmp-dev]}}
 
   # Build on OS X in addition to Linux
+  - env: BUILD=stack ARGS=" "
+    compiler: ": #stack default osx"
+    os: osx
+
   - env: BUILD=stack ARGS="--resolver lts-2"
+    compiler: ": #stack 7.8.4 osx"
     os: osx
 
   - env: BUILD=stack ARGS="--resolver lts-3"
+    compiler: ": #stack 7.10.2 osx"
     os: osx
 
   - env: BUILD=stack ARGS="--resolver lts-5"
+    compiler: ": #stack 7.10.3 osx"
     os: osx
 
   - env: BUILD=stack ARGS="--resolver nightly"
+    compiler: ": #stack nightly osx"
     os: osx
 
   allow_failures:
-  - env: BUILD=cabal GHCVER=head  CABALVER=head
+  - env: BUILD=cabal GHCVER=head  CABALVER=head HAPPYVER=1.19.5 ALEXVER=3.1.7
   - env: BUILD=stack ARGS="--resolver nightly"
 
-before_install:
 
+install:
+# SYMENGINE INSTALL PHASE
+# -----------------------
+# Download and install SymEngine
+- |
+  set -ex
+  export our_install_dir=$HOME/symengine-install
+  mkdir -p $our_install_dir
+  cd $HOME && git clone https://github.com/symengine/symengine.git
+ 
+# Setup C compiler variables
+# The reason we need to do this is because our build system is haskell, so none of the 
+# C variables are set.
+- |
+  if [ `uname` = "Darwin" ]
+  then
+    export CC="clang" && export CXX="clang++" && export CMAKE_C_COMPILER=$CC && export CMAKE_CXX_COMPILER=$CXX
```

Comment:
@isuruf  - I [removed the exports here](https://travis-ci.org/symengine/symengine.hs/jobs/131342044), But the build fails asking for the exports as seen here:

[Link to failure](https://travis-ci.org/symengine/symengine.hs/jobs/131342044#L508)